server/text_generation_server/models/starcoder.py

```diff
@@ -0,0 +1,49 @@
+from loguru import logger
+import torch
+from dataclasses import dataclass
+from typing import List, Optional, Type
+
+from text_generation_server.models import CausalLM
+from text_generation_server.models.causal_lm import CausalLMBatch
+
+
+@dataclass
+class StarCoderCausalLMBatch(CausalLMBatch):
+    past_key_values: Optional[List[torch.Tensor]]
+
+    def detach_kv_cache(self):
+        past_keys = []
+        past_values = []
+        last_dim = int(self.past_key_values[0].size(dim=-1)/2)
+        for key_value in self.past_key_values:
+            past_keys.append(key_value.split((last_dim, last_dim), dim=-1)[0])
+            past_values.append(key_value.split((last_dim, last_dim), dim=-1)[1])
+        del self.past_key_values
+
+        return past_keys, past_values
+
+    def attach_kv_cache(self, past_keys, past_values):
+        self.past_key_values = [
+            torch.cat((key, value), dim=-1) for key, value in zip(past_keys, past_values)]
+
+
+class STARCODER(CausalLM):
+    def __init__(
+        self,
+        model_id: str,
+        revision: Optional[str] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        super(STARCODER, self).__init__(
+            model_id=model_id,
+            revision=revision,
+            dtype=dtype,
+        )
+
+        # Bypasses runtime error "Empty tensor optional" with hpu graphs
+        self.enable_hpu_graph = False
+        logger.warning("Disabling HPU graphs as they are not supported with Starcoder model!")
```

Comment:
This will not work. Model is already initialized and wrapped into HPU graph in a base class constructor. I would suggest to move this before super() call. Please note, that base class constructor will read env variable `ENABLE_HPU_GRAPH`, so you should modify this var.